executor/cook/config.py

```diff
@@ -49,12 +52,24 @@ def initialize_config(environment):
     """Initializes the config using the environment.
     Populates the default values for missing environment variables.
     """
+    executor_id = environment.get('MESOS_EXECUTOR_ID', 'executor')
+    sandbox_directory = environment.get('MESOS_SANDBOX', '')
+    default_progress_output_name = '{}.progress'.format(executor_id)
+    if sandbox_directory:
+        default_progress_output_file = os.path.join(sandbox_directory, default_progress_output_name)
+    else:
+        default_progress_output_file = default_progress_output_name
+
+    progress_output_env_variable = environment.get('EXECUTOR_PROGRESS_OUTPUT_FILE_ENV', 'EXECUTOR_PROGRESS_OUTPUT_FILE')
+    logging.info('Progress location environment variable is {}'.format(progress_output_env_variable))
+    if progress_output_env_variable not in environment:
+        logging.warning('No entry found for {} in the environment'.format(progress_output_env_variable))
+
     max_bytes_read_per_line = int(environment.get('EXECUTOR_MAX_BYTES_READ_PER_LINE', 4 * 1024))
     max_message_length = int(environment.get('EXECUTOR_MAX_MESSAGE_LENGTH', 512))
-    progress_output_name = environment.get('PROGRESS_OUTPUT_FILE', 'stdout')
+    progress_output_name = environment.get(progress_output_env_variable, default_progress_output_file)
```

Comment:
Do we need to allow the user to set this?  What's the use case?

Reply:
The user will set this when they are not happy with the default. E.g. if the progress is being written to `stdout`.